### PR TITLE
chore: Simplified GetRepoHTTPClient function

### PR DESCRIPTION
**Pull request by ls0f**
Signed-off-by: ls0f <lovedboy.tk@qq.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: util/git/client.go [hunk 1]
```diff
@@ -203,16 +203,15 @@ func GetRepoHTTPClient(repoURL string, insecure bool, creds Creds, proxyURL stri
...
 	if insecure {
-		customHTTPClient.Transport = &http.Transport{
-			Proxy: proxyFunc,
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify:   true,
-				GetClientCertificate: clientCertFunc,
-			},
-			DisableKeepAlives: true,
-		}
+		transport.TLSClientConfig.InsecureSkipVerify = true
```
Comment (crenshaw-dev) [inline util/git/client.go:214]: Could we just return early here to avoid the large `else` block?

File: util/git/client.go [hunk 2]
```diff
@@ -223,26 +222,10 @@ func GetRepoHTTPClient(repoURL string, insecure bool, creds Creds, proxyURL stri
...
-		} else {
-			// else no custom certificate stored.
-			customHTTPClient.Transport = &http.Transport{
-				Proxy: proxyFunc,
-				TLSClientConfig: &tls.Config{
-					GetClientCertificate: clientCertFunc,
-				},
-				DisableKeepAlives: true,
-			}
+			transport.TLSClientConfig.RootCAs = certPool
```
Comment (crenshaw-dev) [inline util/git/client.go:225]: Not a requirement, but if you had a minute to update `TestCustomHTTPClient` to cover this block, that would be awesome.  :-)

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Nice optimization! Added a couple nitpicks.

File: util/git/client.go (hunk 1 above)
Comment (ls0f) [reply util/git/client.go:214]: > Could we just return early here to avoid the large `else` block?

I will update :)

**Comment by ls0f** [comment]
@crenshaw-dev  is there any progress please?

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks!

**Pull request merged**